server.js

```diff
@@ -0,0 +1,188 @@
+// DEPENDENCIES
+const express = require('express');
+const logger = require('morgan');
+const cors = require('cors');
+const helmet = require('helmet');
+
+// DATABASE HELPERS
+const userDb = require('./data/helpers/userDb');
+const postDb = require('./data/helpers/postDb');
+const tagDb = require('./data/helpers/tagDb');
+
+// SERVER
+const server = express();
+
+// MIDDLEWARE
+const cruiseControl = (req, res, next) => {
+	req.body.name = req.body.name.toUpperCase();
+	next();
+};
+
+server.use(logger('tiny'), cors(), helmet(), express.json());
+
+// ENDPOINTS
+// get users
+server.get('/api/users', (req, res) => {
+	userDb
+		.get()
+		.then(users => {
+			console.log('\n=== RETURNED USERS: ===\n\n', users, '\n');
+			res.status(200).json(users);
+		})
+		.catch(err => {
+			console.log('\n=== UH OH ===\n', err);
+			res
+				.status(500)
+				.json({ error: 'Users information could not be retrieved.' });
+		});
+});
+
+// get user by id
+server.get('/api/users/:id', (req, res) => {
+	userDb
+		.get(req.params.id)
+		.then(user => {
+			if (!user) {
+				console.log('\n=== NO USER BY THAT ID ===\n');
+				return res
+					.status(404)
+					.json({ message: 'User with specified ID does not exist.' });
+			}
+			console.log('\n=== RETURNED USER: ===\n\n', user, '\n');
+			res.status(200).json(user);
+		})
+		.catch(err => {
+			console.log('\n=== UH OH ===\n\n', err);
+			res
+				.status(500)
+				.json({ error: 'User information could not be retrieved.' });
+		});
+});
+
+// add new user
+//    todo: return message when name already exists
+server.post('/api/users', cruiseControl, (req, res) => {
+	if (!req.body.name) {
+		console.log(`\n=== USER NAME NOT PROVIDED ===\n\n`);
+		return res.status(400).json({
+			errorMessage: 'Please provide name for the user.'
+		});
+	}
+
+	if (req.body.name.length > 128) {
+		console.log(`\n=== USER NAME EXCEEDED 128 CHARACTER LIMIT ===\n\n`);
+		return res.status(400).json({
+			errorMessage: 'User name cannot exceed 128 characters.'
+		});
+	}
+
+	userDb
+		.insert(req.body)
```

Comment:
It is considered a better practice to  destructure instead of just directly adding in req.body. You can do that like this:
```
	const { name } = req.body;
	const newUser = { name };
	userDb
		.insert(newUser)
```
or even:
```
        const newUser = req.body;
	userDb
		.insert(newUser)
```
At least that's what we've been taught. Why? Ryan said it will prevent additional malicious code. I tried testing this by adding a separate field and, at least with SQLite, it will reject the entry as a whole, saying there isn't a field for it. However, also found this: https://stackoverflow.com/questions/19030220/is-it-ok-to-work-directly-on-the-data-in-req-body . 
Ultimately I don't have strong evidence for it atm, but maybe something to consider/look into.